cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/stream/FileStreamAdmin.java

```diff
@@ -247,7 +226,7 @@ public void updateConfig(StreamConfig config) throws IOException {
       // This call will also save the config
       streamCoordinatorClient.changeTTL(originalConfig, config.getTTL());
     }
-    if (originalConfig.getNotificationThresholdMB() != config.getNotificationThresholdMB()) {
+    if (!originalConfig.getNotificationThresholdMB().equals(config.getNotificationThresholdMB())) {
```

Comment:
this change is just to do the right comparison. There will be proper fix for the update config issue after this PR. This PR is a stepping stone to first simplify the class hierarchy before making more changes.